src/ui/screens/Parachains/OverviewScreen.tsx

```diff
@@ -1,74 +1,60 @@
-import React, {useMemo} from 'react';
+import React from 'react';
 import {View, StyleSheet, FlatList} from 'react-native';
 import {Divider, Card, List, Caption, Text, Subheading} from '@ui/library';
-import {formatNumber, BN_ONE, BN_HUNDRED, bnToBn, bnToHex} from '@polkadot/util';
-import type {ParaId} from '@polkadot/types/interfaces';
 import SafeView, {noTopEdges} from '@ui/components/SafeView';
-import {useUpcomingParaIds} from 'src/api/hooks/useUpcomingParaIds';
-import {useParachainIds} from 'src/api/hooks/useParachainIds';
-import {LeasePeriod, useParachainsLeasePeriod} from 'src/api/hooks/useParachainsLeasePeriod';
-import {useParachainProposalIds} from 'src/api/hooks/useParachainProposalIds';
-import {useParachainLeases} from 'src/api/hooks/useParachainLeases';
 import ProgressChartWidget from '@ui/components/ProgressWidget';
-import {useBlockTime} from 'src/api/hooks/useBlockTime';
 import StatInfoBlock from '@ui/components/StatInfoBlock';
 import globalStyles, {standardPadding} from '@ui/styles';
 import {Padder} from '@ui/components/Padder';
-import {useParaEndpoints} from 'src/api/hooks/useParaEndpoints';
-import {BlockTime} from '@ui/components/BlockTime';
 import LoadingView from '@ui/components/LoadingView';
 import {useNavigation} from '@react-navigation/core';
 import {parachainDetailScreen} from '@ui/navigation/routeKeys';
-import {getLeasePeriodString} from 'src/api/utils/parachainLeases';
+import {EmptyView} from '@ui/components/EmptyView';
+import {Parachain, useParachainsOverview} from 'src/api/hooks/useParachainsOverview';
 
 export function ParachainsOverviewScreen() {
-  const {data: leasePeriod, isLoading: isLeasePeriodLoading} = useParachainsLeasePeriod();
-  const {data: parathreadIds} = useUpcomingParaIds();
-  const {data: parachains, isLoading: isLoadingParachains} = useParachainIds();
-  const {data: proposalIds} = useParachainProposalIds();
-  const {timeStringParts: periodRemainderTimeParts} = useBlockTime(leasePeriod?.remainder);
-  const periodRemainder = periodRemainderTimeParts.filter(Boolean).slice(0, 2).join(' ');
-  const {timeStringParts: totalPeriodTimeParts} = useBlockTime(leasePeriod?.length);
-  const totalPeriod = totalPeriodTimeParts.filter(Boolean).slice(0, 2).join(' ');
-  const progressPercent = leasePeriod?.progress
-    .mul(BN_HUNDRED)
-    .div(leasePeriod.length ?? BN_ONE)
-    .toNumber();
-
-  if (isLeasePeriodLoading || isLoadingParachains) {
+  const {parachainsInfo, parachains, loading} = useParachainsOverview();
+  if (loading) {
```

Comment:
```suggestion
  if (loading && !parachainsInfo) {
```
We don't want to display the loading indicator if we already have cached data.